runtime/jcl/common/stdinit.c

```diff
@@ -949,11 +949,16 @@ jint
 completeInitialization(J9JavaVM * vm)
 {
 	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
-	vmFuncs->internalEnterVMFromJNI(vm->mainThread);
-	vmFuncs->sendCompleteInitialization(vm->mainThread, 0, 0, 0, 0);
-	vmFuncs->internalReleaseVMAccess(vm->mainThread);
-
-	if (vm->mainThread->currentException) {
+	J9VMThread *currentThread = vm->mainThread;
+	
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+	vmFuncs->sendCompleteInitialization(currentThread, 0, 0, 0, 0);
+	vmFuncs->internalReleaseVMAccess(currentThread);
+
+	/* ensure ClassLoader.applicationClassLoader updated via system property java.system.class.loader is updated in VM as well */
+	internalInitializeJavaLangClassLoader((JNIEnv*)(currentThread));
```

Comment:
currentException should be checked before this call as well

Reply:
Fixed. Pls have another look.